slycot/tests/test_examples.py

```diff
@@ -6,24 +6,33 @@
 """
 
 from inspect import getmembers, isfunction
-import pytest
+import unittest, io
+from parameterized import parameterized
+from contextlib import redirect_stdout, redirect_stderr
 
 from slycot import examples
 
-examplefunctions = [fun for (fname, fun) in getmembers(examples)
+examplefunctions = [(fname[:-len("_example")], fun)
+                    for (fname, fun) in getmembers(examples)
                     if isfunction(fun) and "_example" in fname]
 
-
-@pytest.mark.parametrize('examplefun', examplefunctions)
-def test_example(examplefun, capsys, recwarn):
-    """
-    Test the examples.
-
-    Test that all the examples work, produce some (unchecked) output but no
-    exceptions or warnings.
-    """
-    examplefun()
-    captured = capsys.readouterr()
-    assert len(captured.out) > 0
-    assert not captured.err
-    assert not recwarn
+class TestExamples(unittest.TestCase):
+
+    @parameterized.expand(examplefunctions)
+    def test_example(self, name, examplefun):
+        #def test_example(examplefun, capsys, recwarn):
+        """
+        Test the examples.
+        
+        Test that all the examples work, produce some (unchecked) output but no
+        exceptions or warnings.
+        """
+        out, err = io.StringIO(), io.StringIO()
+        with redirect_stderr(err), redirect_stdout(out):
+            examplefun()
+        self.assertTrue(len(out.getvalue()) > 0)
+        self.assertFalse(err.getvalue())
+        #assert not recwarn
```

Comment:
It's a pity to lose this warning check, but we can and probably should set a treat-warnings-as-errors flag for *all* the unit tests.